src/azure-cli-core/azure/cli/core/commands/command_operation.py

```diff
@@ -0,0 +1,509 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import argparse
+from six import string_types, get_method_function
+from azure.cli.core import AzCommandsLoader, EXCLUDED_PARAMS
+from knack.introspection import extract_args_from_signature, extract_full_summary_from_signature
+from knack.arguments import CLICommandArgument, ignore_type
+
+
+class BaseCommandOperation:
+    """ Base class of command operation classes """
+
+    def __init__(self, ctx, **kwargs):
+        if not isinstance(ctx, AzCommandsLoader):
+            raise TypeError("'ctx' expected type '{}'. Got: '{}'".format(AzCommandsLoader.__name__, type(ctx)))
+        self.ctx = ctx  # AzCommandsLoader
+        self.cmd = None
+        self.kwargs = kwargs
+        self.client_factory = kwargs.get('client_factory')
```

Comment:
client factory could be in `kwargs`, but it also may be in command type. Could you resolve such scenario?
https://github.com/Azure/azure-cli/blob/6075b8a178c7459c59cddde03c44f987e09cde2e/src/azure-cli/azure/cli/command_modules/network/commands.py#L77

Reply:
That's a very good question. Kwargs is merged in before. I will add commands for it.

https://github.com/Azure/azure-cli/blob/e5eb2dc36e9ae0544b9adcd82e5d40bced0be519/src/azure-cli-core/azure/cli/core/commands/__init__.py#L1255-L1270